### PR TITLE
Feature/typescript 2

### PR DESCRIPTION
**Pull request by jesse-mm**
- Update TypeScript to 2.2.2
- Remove typings dependency
- Fix docs
- Fix lint issue
- Replace rm with rimraf for multi-platform support

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
+  "version": "1.1.0",
   "description": "Manages a config file by merging environments and providing a useful API",
   "main": "dist/index.js",
-  "typings": "dist/index",
   "scripts": {
-    "postinstall": "npm run typings",
     "prepublish": "npm run build",
-    "typings": "typings install",
-    "build": "rm -rf dist && tsc -p ./",
+    "build": "rimraf -rf dist && tsc -p ./",
```
Comment (ThaNarie) [inline package.json]: Can you change this to use shx instead?
https://github.com/shelljs/shx
Adds support for more commands besides "rm", and we also use this in other seng projects.

(don't forget to also update docs/changelog)

File: src/lib/ConfigManager.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 "use strict";
```
Comment (ThaNarie) [inline src/lib/ConfigManager.js]: .js files in src should not be in git (only .ts files)
Could we add them to `.gitignore` and remove them?

File: test/tsconfig.json [hunk 3]
```diff
@@ -7,10 +7,12 @@
 		"removeComments": false,
 		"preserveConstEnums": true,
 		"sourceMap": false,
-		"experimentalDecorators": true
+		"experimentalDecorators": true,
+    "typeRoots": [
```
Comment (ThaNarie) [inline test/tsconfig.json]: Spacing mismatch, according to `.editorconfig` json files should (indeed) have 2 spaces, can you reformat the other lines?

File: tslint.json [hunk 4]
```diff
@@ -30,7 +30,7 @@
     "no-eval": true,
     "no-string-literal": false,
     "no-switch-case-fall-through": false,
-    "no-trailing-comma": true,
+    "trailing-comma": [false],
```
Comment (ThaNarie) [inline tslint.json]: Can we change this to force trailing comma's? it's more in line with the airbnb standard we try to follow.

File: CHANGELOG.MD [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+### Changed
+- Upgraded Typescript 1.8.0 to 2.2.2
+- Remove custom typedoc installation to official one
+
+### Removed
+- Removed typings dependency
+
+### Fixed
+- Fix rm command using rimraf
+- Change rule for eslint (tslint was failing)
```
Comment (ThaNarie) [inline CHANGELOG.MD]: We don't use `eslint` :)

**Review by ThaNarie — CHANGES_REQUESTED** [review]

**Comment by jesse-mm** [comment]
Updated with the requested changes.

**Review by ThaNarie — APPROVED** [review]

**Pull request merged**